### PR TITLE
Release Automation

### PR DESCRIPTION
**Pull request by fuddstar**
new file:   configuration_code/scripts/release_automation/git_clone.
    Clones GIT repository
new file:   configuration_code/scripts/release_automation/git_diff.py
    Allows excludes files to be passed in and then tells you if there is a diff. Excluded files include the files
    that bumpversion will automatically bump and createa  false diff
new file:   configuration_code/scripts/release_automation/ra.py
    Release automation that merges integration otmaster, bymps version and create tags
new file:   configuration_code/scripts/release_automation/ra.sh
    shell version of ra.py

### CODE REVIEW

File: configuration_code/scripts/release_automation/git_clone.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ra_short_git_url = ra_git_repo_tuple[0]
+ra_git_user = args.user
+ra_git_password = args.password
+
+# Clone Repo
+if ra_git_user:
+    ra_clone = Repo.clone_from('https://{0}:{1}@{2}'.format(ra_git_user, ra_git_password, ra_short_git_url),
+                               ra_git_repo_name)
+else:
+    ra_clone = Repo.clone_from(ra_short_git_url, ra_git_repo_name)
```
Comment (ga-autobots) [inline configuration_code/scripts/release_automation/git_clone.py]: short url?

File: configuration_code/scripts/release_automation/git_diff.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+
+"""
+Release Automation Script
+This will:
+  Merge integration into master (assuming those are the name of hte two branches you want to use)
```
Comment (ga-autobots) [inline configuration_code/scripts/release_automation/git_diff.py]: Update Doco - typo the

File: configuration_code/scripts/release_automation/git_clone.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+"""
+Release Automation Script
```
Comment (ga-autobots) [inline configuration_code/scripts/release_automation/git_clone.py]: Try Two Repos combine into ra.py

File: configuration_code/scripts/release_automation/ra.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    # Drop https:// prefix
+    pattern = re.compile('^(https://|http://)', re.IGNORECASE)
+    short_git_url = pattern.sub('', url_dot_git)
+
+    # Get name of git repo for paths
+    git_repo_name = short_git_url.split('/')[-1].split('.')[-2]
+
+    return short_git_url, git_repo_name, url_dot_git
+
+
```
Comment (ga-autobots) [inline configuration_code/scripts/release_automation/ra.py]: Blank Line

File: configuration_code/scripts/release_automation/ra.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+git.merge(ra_branch_integration,
+          '--no-edit')
+logging.info('Merging {0} into {1}'.format(ra_branch_integration, ra_branch_master))
+
+# Checkout Integration
+git.checkout(ra_branch_integration)
+logging.info('Checking out branch: {0}'.format(ra_branch_integration))
+
+# Bump Integration Version
+os.chdir('/'.join([os.getcwd(), ra_git_repo_name]))
```
Comment (ga-autobots) [inline configuration_code/scripts/release_automation/ra.py]: Required?

File: configuration_code/scripts/release_automation/ra.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+bump([ra_bump_level,
+      '--list',
+      '--verbose'])
```
Comment (ga-autobots) [inline configuration_code/scripts/release_automation/ra.py]: Pass in *args?

File: configuration_code/scripts/release_automation/ra.sh [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+## Set Orgin
+git remote add central https://${user}:${password}@${repo} && echo "!! GIT REMOTE ADD CENTRAL"
+
+## Merge to Master
+git fetch central && echo "!! GIT FETCH"                                  # Maybe Remove
+git checkout integration && echo "!! GIT CHECKOUT INTEGRATION"
+git checkout master && echo "!! GIT CHECKOUT MASTER"
+diff=$(git diff master...integration)
+echo ${diff}
+if [ -z ${diff} ]; then
```
Comment (ga-autobots) [inline configuration_code/scripts/release_automation/ra.sh]: Remove -z diff

File: configuration_code/scripts/release_automation/ra.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    """
+    # Ensure .git extension
+    url_dot_git = url + '.git' if url[-4:] != '.git' else url
+
+    # Drop https:// prefix
+    pattern = re.compile('^(https://|http://)', re.IGNORECASE)
+    short_git_url = pattern.sub('', url_dot_git)
+
+    # Get name of git repo for paths
+    git_repo_name = short_git_url.split('/')[-1].split('.')[-2]
```
Comment (fuddstar) [inline configuration_code/scripts/release_automation/ra.py]: split(.)[0]?

File: configuration_code/scripts/release_automation/ra.py (hunk 5 above)
Comment (fuddstar) [inline configuration_code/scripts/release_automation/ra.py]: Moved

**Pull request merged**